### PR TITLE
S4MK3: Software mixer support and default pad layout customisation

### PR DESCRIPTION
**Pull request by acolombier**
This addresses two feature requested raised on Zulip

UI change:

_Before:_
image

_After:_
image

### CODE REVIEW

File: res/controllers/Traktor Kontrol S4 MK3.hid.xml [hunk 1]
```diff
@@ -199,6 +214,33 @@
                     When shifting decks, the mixer controls microphones or auxiliary lines. If both a microphone and an auxiliary are configured on the same channel, the mixer will prioritize controlling the auxiliary.
                 </description>
             </option>
+            <option
+                variable="softwareMixerMain"
+                type="boolean"
+                default="false"
+                label="Main gain is handled by the Mixxx built-in mixer">
+                <description>
+                    When enable, the master potentiometer on top right column of the mixer will drive the main gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```
Comment (acolombier) [inline res/controllers/Traktor Kontrol S4 MK3.hid.xml:223]: Not sure if the word `master` is problematic here - the device is physically labelled with `Master` so if it is, perhaps I can put `"` around it or the `:hwbtn:` marker?

**Comment by acolombier** [comment]
Really annoying that the `pre-commit` don't seem to catch the required changes locally (same version, running the same command than the CI), but I guess `eslint` being in Javascript, it is expected to have an unexpected and random behaviour depending of the weather outside :sweat_smile:

**Comment by acolombier** [comment]
Merge?

File: res/controllers/Traktor Kontrol S4 MK3.hid.xml (hunk 1 above)
Comment (Swiftb0y) [inline res/controllers/Traktor Kontrol S4 MK3.hid.xml:223]:
```suggestion
                    When enabled, the master potentiometer on top right column of the mixer will drive the main gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```

File: res/controllers/Traktor Kontrol S4 MK3.hid.xml [hunk 2]
```diff
@@ -199,6 +214,33 @@
...
+                </description>
+            </option>
+            <option
+                variable="softwareMixerBooth"
+                type="boolean"
+                default="false"
+                label="Booth gain is handled by the Mixxx built-in mixer">
+                <description>
+                    When enable, the booth potentiometer on right column of the mixer will drive the booth gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```
Comment (Swiftb0y) [inline res/controllers/Traktor Kontrol S4 MK3.hid.xml:232]:
```suggestion
                    When enabled, the booth potentiometer on right column of the mixer will drive the booth gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```

File: res/controllers/Traktor Kontrol S4 MK3.hid.xml [hunk 3]
```diff
@@ -199,6 +214,33 @@
...
+                </description>
+            </option>
+            <option
+                variable="softwareMixerHeadphone"
+                type="boolean"
+                default="false"
+                label="Headphone gain and cue mix is handled by the Mixxx built-in mixer">
+                <description>
+                    When enable, the headphone volume and cue potentiometers on bottom right column of the mixer will drive the headphone controls of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```
Comment (Swiftb0y) [inline res/controllers/Traktor Kontrol S4 MK3.hid.xml:241]:
```suggestion
                    When enabled, the headphone volume and cue potentiometers on bottom right column of the mixer will drive the headphone controls of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```

File: res/controllers/Traktor Kontrol S4 MK3.hid.xml [hunk 4]
```diff
@@ -199,6 +214,33 @@
                     When shifting decks, the mixer controls microphones or auxiliary lines. If both a microphone and an auxiliary are configured on the same channel, the mixer will prioritize controlling the auxiliary.
                 </description>
             </option>
+            <option
+                variable="softwareMixerMain"
+                type="boolean"
+                default="false"
+                label="Main gain is handled by the Mixxx built-in mixer">
+                <description>
+                    When enabled, the master potentiometer on top right column of the mixer will drive the main gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```
Comment (ywwg) [inline res/controllers/Traktor Kontrol S4 MK3.hid.xml:223]:
```suggestion
                    When enabled, the master potentiometer on top right column of the mixer will drive both the main gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but could result in undesired clipping if enabled when using the hardware audio outputs.
```

File: res/controllers/Traktor Kontrol S4 MK3.hid.xml [hunk 5]
```diff
@@ -199,6 +214,33 @@
...
+                </description>
+            </option>
+            <option
+                variable="softwareMixerBooth"
+                type="boolean"
+                default="false"
+                label="Booth gain is handled by the Mixxx built-in mixer">
+                <description>
+                    When enabled, the booth potentiometer on right column of the mixer will drive the booth gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```
Comment (ywwg) [inline res/controllers/Traktor Kontrol S4 MK3.hid.xml:232]:
```suggestion
                    When enabled, the booth potentiometer on right column of the mixer will drive both the booth gain of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but could result in undesired clipping if enabled when using the hardware audio outputs.
```

File: res/controllers/Traktor Kontrol S4 MK3.hid.xml [hunk 6]
```diff
@@ -199,6 +214,33 @@
...
+                </description>
+            </option>
+            <option
+                variable="softwareMixerHeadphone"
+                type="boolean"
+                default="false"
+                label="Headphone gain and cue mix is handled by the Mixxx built-in mixer">
+                <description>
+                    When enabled, the headphone volume and cue potentiometers on bottom right column of the mixer will drive the headphone controls of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but will create conflict if enabled when using the hardware outputs.
```
Comment (ywwg) [inline res/controllers/Traktor Kontrol S4 MK3.hid.xml:241]:
```suggestion
                    When enabled, the headphone volume and cue potentiometers on bottom right column of the mixer will drive both the headphone controls of the Mixxx internal mixer as well as the hardware built-in mixer in the device. This is useful if you aren't relying on the hardware output, but could create undesirable clipping if enabled when using the hardware audio outputs.
```

**Review by ywwg — APPROVED** [review]
small suggestion for wording, otherwise looks good

**Comment by acolombier** [comment]
As the [manual PR](https://github.com/mixxxdj/manual/pull/636) has been merged now, are we happy to merge this one too?

**Comment by acolombier** [comment]
Can someone from the Core Team add the 2.5 milestone on this PR? The manual for this was merged so that will have to go with it (or need revert on manual)

**Comment by ywwg** [comment]
I should be able to quickly test this on my hardware and merge today if it's good

**Pull request merged**